### PR TITLE
[CSL-746] General improvements

### PR DESCRIPTION
**Pull request by esezen**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,6 @@ help:
 .PHONY: help Makefile
 
 docs: Makefile
-	pipenv run sphinx-apidoc -o source constructor_io
```
Comment (esezen) [inline Makefile:13]: This is to hide helpers module. We will create the rst files ourselves

File: constructor_io/helpers/utils.py [hunk 2]
```diff
@@ -78,7 +79,7 @@ def create_shared_query_params(options, parameters, user_parameters):
                 for key, value in filters.items():
                     query_params[f'filters[{key}]'] = value
             else:
-                raise Exception('filters must be a dictionary')
+                raise ConstructorException('filters must be a dictionary')
```
Comment (esezen) [inline constructor_io/helpers/utils.py:82]: Use ConstructorException everywhere so users can catch specific error

File: constructor_io/modules/autocomplete.py [hunk 3]
```diff
@@ -5,20 +5,21 @@
...
 
+from constructor_io.helpers.exception import ConstructorException
 from constructor_io.helpers.utils import (clean_params, create_auth_header,
                                           create_request_headers,
                                           create_shared_query_params,
                                           throw_http_exception_from_response)
 
 
-def create_autocomplete_url(query, parameters, user_parameters, options):
+def _create_autocomplete_url(query, parameters, user_parameters, options):
```
Comment (esezen) [inline constructor_io/modules/autocomplete.py:15]: Mark methods as "weak internal use" so they are private and do not appear in docs

Comment (sblaurock) [reply constructor_io/modules/autocomplete.py:15]: Love these comments - thank you!

File: tests/modules/test_browse.py [hunk 4]
```diff
@@ -657,7 +659,7 @@ def test_get_browse_facets_with_invalid_fmt_options():
 def test_get_browse_facets_with_invalid_api_key():
     '''Should return a response with invalid api_key'''
 
-    with raises(Exception, match=r'We have no record of this key. You can find your key at app.constructor.io/dashboard.'): # pylint: disable=line-too-long
+    with raises(HttpException, match=r'We have no record of this key. You can find your key at app.constructor.io/dashboard.'): # pylint: disable=line-too-long
```
Comment (sblaurock) [inline tests/modules/test_browse.py:662]: Intentional? Should this be ConstructorException?

**Review by sblaurock — APPROVED** [review]
One comment but I believe everything looks great! Solid work.

**Pull request merged**